apps/prairielearn/elements/pl-order-blocks/dag_checker.py

```diff
@@ -185,7 +202,7 @@ def lcs_partial_credit(submission, depends_graph, group_belonging):
     :return: edit distance from the student submission to some correct solution
     """
     graph = dag_to_nx(depends_graph, group_belonging)
-    trans_clos = nx.algorithms.dag.transitive_closure(graph)
+    trans_clos = nx.algorithms.dag.transitive_closure(graph)  # type: ignore
```

Comment:
Does the typechecker complain without the ignore statement here? Seems a little weird that it would.

Reply:
```
/PrairieLearn/apps/prairielearn/elements/pl-order-blocks/dag_checker.py:205:32 - error: "dag" is not a known member of module "networkx.algorithms" (reportGeneralTypeIssues)
```

Reply:
Ya, its weird. I don't know what's going on here.

Reply:
I think that this might require doing import network.algorithms. Otherwise, I think we can set hype checking to ignore networkx since it has typing set up weird.

Reply:
that doesn't work. But if I do `from networkx.algorithms import dag`, then change this line to `dag.transitive_closure...` that does work. 

I really don't like doing an import like that to only use it once though.

Reply:
I think it's technically a separate import from just networkx. Same as the lxml double imports.

Reply:
Its rather annoying because the code runs just fine the other way. Why would import resolution work one way for type checking and a different way for executing?

Reply:
I think it can actually be imported from networkx directly: https://networkx.org/documentation/stable/reference/algorithms/generated/networkx.algorithms.dag.transitive_closure.html#networkx.algorithms.dag.transitive_closure

Reply:
oh that's much nicer. thanks.